### PR TITLE
Lex

### PR DESCRIPTION
**Pull request by NolanRapp**
passing all tests !!!

### CODE REVIEW

File: .vscode/settings.json
Comment (EsotericSquishyy) [inline .vscode/settings.json:1]: Delete this and add this file to the .gitignore so it doesn't show up again

File: README.md
Comment (EsotericSquishyy) [inline README.md:1]: Not sure why the README was wiped

File: src/lex.cpp [hunk 1]
```diff
@@ -24,35 +41,84 @@ std::vector<token> lex() {
...
-            else if(isdigit(i)){
-                placeholder = "/0";
-                bool decimal = false;
-                placeholder += i;
-                while (std::cin >> i && (isdigit(i) || i == '.')){
-                    if (i == '.') {
-                        if (decimal){
-                            //error
+            //if number input or decimal input, check validity:
+            else if (isdigit(i) || i == '.') {
```
Comment (EsotericSquishyy) [inline src/lex.cpp]: Don't need to check if the char is '.'
If it is it should go to the else statement that exits the program with an error.
All code related to checking a '.' before the while statement is not needed

File: src/lex.cpp [hunk 2]
```diff
@@ -24,35 +41,84 @@ std::vector<token> lex() {
...
+                bool decimal = (i == '.'); // sets boolean equal to true if i == '.'
+                std::string placeholder(1, i); // this converts a char (1) to a string
+                int startingColumn = column;
+
+                // checking for leading decimal
+                if (i == '.' && !(isdigit(placeholder.back()))) {
```
Comment (EsotericSquishyy) [inline src/lex.cpp]: Not needed, see previous comment

File: src/lex.cpp [hunk 3]
```diff
@@ -24,35 +41,84 @@ std::vector<token> lex() {
...
 
-    createend(output);
-    printtokens(output);
-}
\ No newline at end of file
+int main() {
+
+	Lexer lexer;
+	lexer.lex();
+	lexer.printTokens();
```
Comment (EsotericSquishyy) [inline src/lex.cpp]: weird indentation

File: src/lex.cpp [hunk 4]
```diff
@@ -24,35 +41,84 @@ std::vector<token> lex() {
...
+            }
         }
-    }
+        // if input is space, add column:
+        column++;
+    } 
+    createEnd(lexTokens, line, column);
+}
+
+
```
Comment (EsotericSquishyy) [inline src/lex.cpp:116]: I think we should do 3 newlines between functions of same namespace and 6 lines between different namespace functions

File: README.md
Comment (chloeandersen-ucsb) [reply README.md:1]: i changed it a ton in our parse branch, so I figured I'd wipe it here so we don't get any merge conficts when pulling from main in parse branch
Let me know if that doesn't make sense though

**Pull request merged**